src/twisted/names/dns.py

```diff
@@ -395,26 +425,12 @@ def str2time(s):
     @return: an L{int} giving the interval represented by the string C{s}, or
         whatever C{s} is if it is not a string.
     """
-    suffixes = (
-        ("S", 1),
-        ("M", 60),
-        ("H", 60 * 60),
-        ("D", 60 * 60 * 24),
-        ("W", 60 * 60 * 24 * 7),
-        ("Y", 60 * 60 * 24 * 365),
-    )
     if isinstance(s, bytes):
-        s = s.decode("ascii")
```

Comment:
mypy doesn't like type-punning this s: str | bytes | int | None into a str so I used a new function